ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt

```diff
@@ -145,13 +145,27 @@ constructor(
   }
 
   suspend fun enableLocationLockAndGetUpdates() {
-    permissionsManager.obtainPermission(Manifest.permission.ACCESS_FINE_LOCATION)
-    settingsManager.enableLocationSettings(FINE_LOCATION_UPDATES_REQUEST)
-    enableLocationLock()
-    locationManager.requestLocationUpdates()
+    try {
+      permissionsManager.obtainPermission(Manifest.permission.ACCESS_FINE_LOCATION)
+      settingsManager.enableLocationSettings(FINE_LOCATION_UPDATES_REQUEST)
+      enableLocationLock()
+      locationManager.requestLocationUpdates()
+    } catch (throwable: ApiException) {
+      val statusCode = throwable.statusCode
+      if (statusCode == SETTINGS_CHANGE_UNAVAILABLE) {
+        Timber.e(
+          throwable,
+          "User is offline, so fallback to user's current permission, which may also fail."
+        )
+        enableLocationLock()
+        locationManager.requestLocationUpdates()
+      } else {
+        throw throwable
```

Comment:
Instead of rethrowing, should we directly call `handleRequestLocationUpdateFailed`? Then we can get rid of the try-catch block in `CaptureLocationTaskMapFragment.kt` Line 57-61

Reply:
Unified all the try/catch into the `enableLocationLockAndGetUpdates()` so that we can call it without fear of throwing errors that will crash the app. The error message will now always funnel through the location lock icon failure handler, which is pleasant.